DM-35937: Get the composite dataset type from the component dataset type

timj opened the pull request:
There is no need to ask registry for this (assuming the component
dataset type has been properly configured with a parent storage
class). This also prevents look ups into registry for non-existent
dataset types which do not occur in the else branch (which always
uses the given dataset type).

@kfindeisen can you see if this works for you?

## Checklist

- [x] ran Jenkins
- [ ] added a release note for user-visible changes to `doc/changes`

timj committed:
Get the composite dataset type from the component dataset type

There is no need to ask registry for this (assuming the component
dataset type has been properly configured with a parent storage
class). This also prevents look ups into registry for non-existent
dataset types which do not occur in the else branch (which always
uses the given dataset type).

kfindeisen commented:
Yes, it works, both in the demo and in our original `prompt_prototype` case. Thanks for the quick response! 🙂

timj committed:
Modify the "no registered dataset type" message to match other places

Otherwise the doomed_by message you get from a dataset type name
not being registered differs from the doomed_by message you get
when a full dataset type is used but not known.

timj committed:
Add some tests for the findDatasets code

Confirmed that this fixes the problem reported on the ticket.
